File: messenger/src/org/thoughtcrime/securesms/jobmanager/JobLogger.java
package org.thoughtcrime.securesms.jobmanager;

import androidx.annotation.NonNull;
import android.text.TextUtils;

public class JobLogger {

  public static String format(@NonNull Job job, @NonNull String event) {
    return format(job, "", event);
  }

  public static String format(@NonNull Job job, @NonNull String extraTag, @NonNull String event) {
    String id                  = job.getId();
    String tag                 = TextUtils.isEmpty(extraTag) ? "" : "[" + extraTag + "]";
    long   timeSinceSubmission = System.currentTimeMillis() - job.getParameters().getCreateTime();
    int    runAttempt          = job.getRunAttempt() + 1;
    String maxAttempts         = job.getParameters().getMaxAttempts() == Job.Parameters.UNLIMITED ? "Unlimited"
                                                                                                  : String.valueOf(job.getParameters().getMaxAttempts());
    String lifespan            = job.getParameters().getLifespan() == Job.Parameters.IMMORTAL ? "Immortal"
                                                                                              : String.valueOf(job.getParameters().getLifespan()) + " ms";
    return String.format("[%s][%s]%s %s (Time Since Submission: %d ms, Lifespan: %s, Run Attempt: %d/%s)",
                         id, job.getClass().getSimpleName(), tag, event, timeSinceSubmission, lifespan, runAttempt, maxAttempts);
  }
}

File: app/src/main/java/org/thoughtcrime/securesms/jobmanager/JobLogger.kt
package org.thoughtcrime.securesms.jobmanager

import android.text.TextUtils

/**
 * Provides utilities to create consistent logging for jobs.
 */
object JobLogger {

  @JvmStatic
  fun format(job: Job, event: String): String {
    return format(job, "", event)
  }

  @JvmStatic
  fun format(job: Job, extraTag: String, event: String): String {
    val id = job.id
    val tag = if (TextUtils.isEmpty(extraTag)) "" else "[$extraTag]"
    val timeSinceSubmission = System.currentTimeMillis() - job.parameters.createTime
    val runAttempt = job.runAttempt + 1
    val maxAttempts = if (job.parameters.maxAttempts == Job.Parameters.UNLIMITED) "Unlimited" else job.parameters.maxAttempts.toString()
    val lifespan = if (job.parameters.lifespan == Job.Parameters.IMMORTAL) "Immortal" else job.parameters.lifespan.toString() + " ms"

    return "[JOB::$id][${job.javaClass.simpleName}]$tag $event (Time Since Submission: $timeSinceSubmission ms, Lifespan: $lifespan, Run Attempt: $runAttempt/$maxAttempts, Queue: ${job.parameters.queue})"
  }
}
